client/client.go

```diff
@@ -60,31 +77,54 @@ func NewClient(ctx context.Context, apiKey string, orgName string, env string) *
 
 	return &Client{
 		apiKey:      apiKey,
+		orgName:     orgName,
 		baseURL:     baseURL,
-		client:      http.DefaultClient,
+		userAgent:   userAgent,
+		rateLimiter: rate.NewLimiter(rate.Limit(DefaultRateLimitPerSecond), 1),
+		client: &retryablehttp.Client{
+			HTTPClient:   http.DefaultClient,
+			CheckRetry:   checkHTTPRetry,
+			RetryWaitMin: 3 * time.Second,
+			Backoff:      retryablehttp.DefaultBackoff,
+			RetryMax:     DefaultRetryMax,
+		},
 		contentType: "application/vnd.api+json",
 	}
 }
 
+// checkHTTPRetry inspects HTTP errors from the Lightstep API for known transient errors
+func checkHTTPRetry(_ context.Context, resp *http.Response, err error) (bool, error) {
+	if resp.StatusCode == http.StatusInternalServerError {
+		return true, nil
+	}
+	return false, nil
+}
+
 // CallAPI calls the given API and unmarshals the result to into result.
 func (c *Client) CallAPI(httpMethod string, suffix string, data interface{}, result interface{}) error {
```

Comment:
I think the correct thing to do here is to pass in the context as first parameter...

Reply:
Just did a big pass on cleaning up context, now passed to each API call method